### PR TITLE
GDPR: Anonymize IP Address in tracking logs

### PR DESCRIPTION
**Pull request by kabirkhan**
#### What does this PR do? Please provide some context

Anonymizes user IP address in logs if SQUELCH_PII_IN_LOGS = True in settings.FEATURES

#### Where should the reviewer start?

common/djangoapps/track/middleware.py

#### How can this be manually tested? (brief repro steps)

set SQUELCH_PII_IN_LOGS in FEATURES section of lms.env.json and cms.env.json

#### Definition of done:
- [ ] Title of the pull request is clear and informative
- [ ] Add pull request hyperlink to relevant TFS items
- [ ] For large or complex change: schedule an in-person review session
- [ ] This change has appropriate test coverage

#### Reminders BEFORE merging
1. Get at least two approvals
1. If you're merging from a feature branch into the development branch then "flatten" or "squash" commits
1. If merging from the development branch into master (or porting changes from upstream) then use github's UI to get review feedback, but use the git command line interface to complete the actual merge.

#### Reminders AFTER merging
1. Delete the remote feature branch
1. Resolve relevant TFS items
1. (reverse merge) If you merged from the development branch into master then check to see if there are any changes in master that can be merged down to the development branch (like hotfixes, etc). In this case, use github's UI for feedback and the git command line interface for the actual merge.

[//]: # ( todo: If you merged into development branch then verify change in our "rolling deployment" environment. Then notify stakeholders interested in or involved with the change )

[//]: # ( fyi: This content was heavily inspired by )
[//]: # ( 1 Our team's policies and processes )
[//]: # ( 2 https://github.com/sprintly/sprint.ly-culture/blob/master/pr-template.md )
[//]: # ( 3 The book "The Checklist Manifesto: How to Get Things Right" by Atul Gawande )
[//]: # ( 4 https://github.com/Azure/azure-event-hubs/blob/master/.github/PULL_REQUEST_TEMPLATE.md )

### CODE REVIEW

File: common/djangoapps/track/views/__init__.py [hunk 1]
```diff
@@ -32,12 +32,56 @@ def _get_request_header(request, header_name, default=''):
         return default
 
 
+def get_request_ip(request):
+    """Wrapper for helper method to get request ip."""
+    return _get_request_ip(request)
+
+
 def _get_request_ip(request, default=''):
```
Comment (sdolenc) [inline common/djangoapps/track/views/__init__.py]: nit: it looks like you could refactor _qet_request_ip into using three return statements (or even one). probably not worth the effort, but i prefer having the fewest number of exit points on a function for readability, debugging, and unit testing

File: common/djangoapps/track/views/__init__.py [hunk 2]
```diff
@@ -32,12 +32,56 @@ def _get_request_header(request, header_name, default=''):
...
+            return default
+
+
+def _get_anonymous_ip(ip_address_str):
+    """Helper method to obbuscate the last two octets of an ip address"""
+    try:
+        ip_comps = ip_address_str.split('.')
+        first, second = '.'.join(ip_comps[0:2]), '.'.join(ip_comps[2:])
+        second_out = ''
+        for char in second:
```
Comment (sdolenc) [inline common/djangoapps/track/views/__init__.py]: i'd prefer we always output three x's or one x. for example
1.2.xxx.xxx
or
1.2.x.x

Comment (sdolenc) [reply common/djangoapps/track/views/__init__.py]: yeah. i'd delete the loop and just do this '{}.{}.x.x'.format(first, second_out)

Comment (kabirkhan) [reply common/djangoapps/track/views/__init__.py]: The idea here is from a BI perspective. 
As a Data Scientist, I'd like the ability to filter on IPs considering 1.2.xxx.xxx and 1.2.xx.xx as different IPs. Provides anonymity but with the most flexibility.

**Comment by kabirkhan** [comment]
Yes they will, this is more of WIP pull

File: common/djangoapps/track/tests/test_middleware.py [hunk 3]
```diff
@@ -93,6 +93,24 @@ def test_default_request_context(self):
             'client_id': None,
         })
 
+    @override_settings(FEATURES={'SQUELCH_PII_IN_LOGS': True})
```
Comment (kevsers) [inline common/djangoapps/track/tests/test_middleware.py]: do we have that PR ready for oxa-tools already or it would follow this one?

Comment (kabirkhan) [reply common/djangoapps/track/tests/test_middleware.py]: This setting is already enabled by default.

**Pull request merged**